src/dune_engine/dep_path.ml

```diff
@@ -93,6 +93,17 @@ let is_loc_none loc =
   | None -> true
   | Some loc -> Loc.is_none loc
 
+(* Similar to [output_starts_with_location] in process.ml but operating on
+   [Pp.t]. *)
+let message_starts_with_location msg =
+  (* The implementation is heavy handed but it doesn't seem worth optimising
+     now. Indeed, we have been discussing the idea of extracting the location
+     from command run by Dune more systematically, so it's likely that this code
+     will go away eventually. *)
+  String.is_prefix
+    (Format.asprintf "%a" Pp.to_fmt (User_message.pp msg))
+    ~prefix:"File "
```

Comment:
Might be worth including the quote character, i.e.:

```suggestion
    ~prefix:"File \""
```

so that this doesn't catch messages like "File not found". Not sure if this can actually happen though.

Reply:
I copied what we do in process.ml for now, but agreed that matching the quote would probably be better. I expect that at some point @rgrinberg and/or @cwong-ocaml might want to rework how we extract locations from command output to consolidate things on the RPC/editor side, so I'll leave things as it for now.